main.py

```diff
@@ -2152,18 +2156,14 @@ def do_request(self, prefix, locations):
 
     def format_completion(self, item) -> 'Tuple[str, str]':
         # Sublime handles snippets automatically, so we don't have to care about insertTextFormat.
-        label = item.get("label")
-        detail = item.get("detail")
-        kind = item.get("kind")
+        label = item["label"]
+        detail = item.get("detail") if show_completion_details else None
         if not detail:
-            if kind is not None:
-                detail = completion_item_kind_names[kind]
-        insertText = item.get("insertText", None)
-        if not insertText:
-            insertText = label
+            detail = completion_item_kind_names[item.get("kind", CompletionItemKind.Unknown)]
+        insertText = item.get("insertText") or label
         if insertText[0] == '$':  # sublime needs leading '$' escaped.
             insertText = '\$' + insertText[1:]
-        return "{}\t{}".format(label, detail) if detail else label, insertText
+        return "{} \t {}".format(label, detail), insertText
```

Comment:
Why the extra spaces between the tab character?

Reply:
> The label and the details of a completion item returned by format_completion() are separated by \t rather than \t as ST tends to display no space between them if the details string is too long otherwise.

Reply:
Whoops, should have read all of your paragraphs! 👍